src/Corporation/Corporation.ts

```diff
@@ -109,6 +111,11 @@ export class Corporation {
       const gameCycles = marketCycles * corpConstants.gameCyclesPerCorpStateCycle;
       this.storedCycles -= gameCycles;
 
+      // Handle "nextCycle" resolvers at the start of this cycle
+      while (this.resolvers.length > 0) {
+        this.resolvers.shift()?.(state);
+      }
```

Comment:
Same comment for all these loops:
This is needlessly O(n^2), since it's O(n) for each shift.
```suggestion
      for (const res of this.resolvers) {
        res(state);
      }
      this.resolvers.length = 0;
```

Reply:
presumably the best performance would be 

```js
for (const resolver of resolvers) {
  resolver();
}
resolvers.length = 0;
```

but `pop()` seems more sane

Reply:
The access of `resolvers` inside `this` will essentially be hoisted out by the JIT compiler... (not exactly, but essentially), so there's no need to do that explicitly unless this is the tightest of loops and profiling shows it matters.

Optional: However, there's a subtle issue with `pop()`: It changes the behavior from being queue-like to stack-like, i.e. the first awaiter on `nextCycle` will become the last to be resolved. This is the opposite of the typical behavior of the microtask queue (that's why it's called a queue), and thus (IMO) against the expectations of users.

Although there's countless ways to do this in JS, I think the `for..of` loop is the clearest and most correct way for this situation.

Reply:
Ah, it looks like the `length = 0` pattern is already in this code base a lot.

I am running into some trouble with saving and loading data with the `for ... of` form, that I think `shift()?()` and `pop()?()` were avoiding. Is there a good way to block these resolvers from being included in save data so that they don't have to be checked at every access?